src/Dapr.AspNetCore/CloudEventsMiddleware.cs

```diff
@@ -70,8 +70,16 @@ private async Task ProcessBodyAsync(HttpContext httpContext, string charSet)
             string originalContentType;
             string contentType;
 
+            // Check whether to use data or data_base64 as per https://github.com/cloudevents/spec/blob/v1.0.1/json-format.md#31-handling-of-data
+            var isDataSet = json.TryGetProperty("data", out var data);
+            var isBinaryDataSet = json.TryGetProperty("data_base64", out var binaryData);
+
+            if(isDataSet && isBinaryDataSet)
+            {
+                throw new DaprException("Both, data and data_base64 fields set in the Cloudevents envelope. Invalid request");
```

Comment:
This should not be an exception. This should mark the request as an HTTP 400 and return. An exception turns into a 500 - a lot of teams using ASP.NET Core in production would treat a 500 as an incident to investigate, so you don't want to let bad data sent by a client turn into a 500.

```C#
httpContext.Response.StatusCode = 400;
return
```